JSUserRegistrationService/src/main/java/com/example/jsuserregistrationservice/AccountInformation.java

```diff
@@ -0,0 +1,42 @@
+package com.example.jsuserregistrationservice;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import javax.validation.constraints.Pattern;
+import org.springframework.data.mongodb.core.index.Indexed;
+
+/**
+ * Copyright (c) 2016, 2017, Jumia.
+ */
+
+public class AccountInformation {
+
+    @NotNull
+    @Pattern(regexp = "^[a-z0-9_-]{4,10}$", message = "Username should be between 4 and 10 characters and contain alphanumeric characters "
+        + "and underscores only")
+    @Indexed(unique = true)
+    private String userName;
+
+    @NotNull
+    @Pattern(regexp = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{4,}$", message = "Password Should contain at least one "
```

Comment:
This regex enforces the required constraints plus at least a lower case letter. Although it does not 100% matches the contract specified, its much more similar to a real world scenario.